octree/include/pcl/octree/impl/octree2buf_base.hpp

```diff
@@ -60,7 +60,7 @@ Octree2BufBase<LeafContainerT, BranchContainerT>::~Octree2BufBase()
 {
   // deallocate tree structure
   deleteTree();
-  delete (root_node_);
+  root_node_.reset();
```

Comment:
This shouldn't be necessary I think: the memory will be automatically deleted by the `std::unique_ptr` destructor.